Mathlib/Tactic/Linter/CasesLinter.lean

```diff
@@ -0,0 +1,53 @@
+/-
+Copyright (c) 2025 Jeremy Tan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jeremy Tan, Damiano Testa
+-/
+import Lean.Elab.Command
+import Mathlib.Tactic.Cases
+
+/-!
+# The "cases" linter
+
+The "cases" linter flags uses of the `cases'` tactic, which is a backward-compatible version of
+Lean 3's `cases` tactic. Unlike `obtain`, `rcases` and Lean 4's `cases`, variables introduced by
+`cases'` are not required to be separated by case, which hinders readability.
+
+This linter is an incentive to discourage uses of `cases'`, without being a ban.
+-/
+
+open Lean Elab
+
+namespace Mathlib.Linter.Style
+
+/-- The "cases" linter flags uses of the `cases'` tactic, which is a backward-compatible version of
+Lean 3's `cases` tactic. Unlike `obtain`, `rcases` and Lean 4's `cases`, variables introduced by
+`cases'` are not required to be separated by case, which hinders readability.
+-/
+register_option linter.style.cases : Bool := {
+  defValue := false
+  descr := "enable the cases linter"
+}
+
+/-- `getCases' t` returns all usages of the `cases'` tactic in the input syntax `t`. -/
+partial
+def getCases' : Syntax → Array Syntax
+  | stx@(.node _ kind args) =>
+    let rargs := (args.map getCases').flatten
+    if kind == ``Mathlib.Tactic.cases' then rargs.push stx else rargs
```

Comment:
```suggestion
    if kind == `Mathlib.Tactic.cases' then rargs.push stx else rargs
```
This does the matching, without checking that the name exists.

The test file can still import the `cases'` linter, though.

Note: I'm not at a computer, so I cannot really test this.

Reply:
Committed the change.